dtmi/osconfig/configuration-1.json

```diff
@@ -0,0 +1,109 @@
+{
+  "@context": "dtmi:dtdl:context;2",
+  "@id": "dtmi:osconfig:configuration;1",
+  "@type": "Interface",
+  "displayName": "Configuration",
+  "description": "Provides functionality to remotely manage OSConfig configuration on device",
+  "contents": [
+    {
+      "@type": "Property",
+      "name": "desiredConfiguration",
+      "schema": {
+        "@type": "Object",
+        "fields": [
+          {
+            "name": "refreshInterval",
+            "schema": "integer"
+          },
+          {
+            "name": "localManagementEnabled",
+            "schema": "boolean"
+          },
+          {
+            "name": "fullLoggingEnabled",
+            "schema": "boolean"
+          },
+          {
+            "name": "commandLoggingEnabled",
+            "schema": "boolean"
+          },
+          {
+            "name": "iotHubProtocol",
+            "schema": {
+              "@type": "Enum",
+              "valueSchema": "integer",
+              "enumValues": [
+                {
+                  "name": "auto",
+                  "enumValue": 0
+                },
+                {
+                  "name": "mqtt",
+                  "enumValue": 1
+                },
+                {
+                  "name": "mqttWebSocket",
+                  "enumValue": 2
+                }
+              ]
+            }
+          }
+        ]
+      },
+      "writable": true
+    },
+    {

```

Comment:
Why not use an object for all reported properties (here and in MIM)? 

This would significantly cut down on the number of `MmiGet` calls to the module (configuration may grow larger in the future). Also, could allow for the reported object to match 1:1 with `osconfig.json`.

Reply:
This MIM was reviewed and signed off earlier in team. 

Reason is that we want to use flatten models where it is possible. The configuration settings can be queried separately on their own fine. However when applying new settings, osconfig needs to be restarted. We do not want if somebody enables full and command logging to restart osconfig twice, for example. Also, some of the settings can be grouped, if not now, in the future (the problem that desired have for HostName, Ztsi, etc).